### PR TITLE
Return Formulas.Tools to netstandard2.0

### PR DESCRIPTION
**Pull request by tehcrashxor**
In commit [1c9b22c](https://github.com/microsoft/PowerApps-Tooling/commit/1c9b22cef15b0da9d8afbdb57aebaac3252807ed) back in Dec 2023, all projects in the repo were set to use the same TargetFramework, all sourced from a single shared file.

This accidentally effected the Formulas.Tooling lib, which needs to remain on netstandard2.0 for use in both .NET and .NET Framework versions of PAC CLI.

This PR switches that library back to netstandard2.0, and reverts a few changes since that point for new syntax net8.0 which is not available to netstandard2.0.

### CODE REVIEW

**Review by petrochuk — APPROVED** [review]

File: src/PAModel/Microsoft.PowerPlatform.Formulas.Tools.csproj [hunk 1]
```diff
@@ -49,8 +51,4 @@
     <PackageReference Include="System.Text.Encodings.Web" Version="6.0.0" />
     <PackageReference Include="YamlDotNet" Version="$(YamlDotNetVersion)" />
   </ItemGroup>
-    
-  <ItemGroup>
-    <ProjectReference Include="..\Persistence\Microsoft.PowerPlatform.PowerApps.Persistence.csproj" />
```
Comment (joem-msft) [inline src/PAModel/Microsoft.PowerPlatform.Formulas.Tools.csproj:54]: oh nice! glad the dependency isn't actually necessary.

File: src/PAModel/Serializers/MsAppSerializer.cs [hunk 2]
```diff
@@ -57,7 +57,10 @@ private static string AsString(ZipArchiveEntry entry)
 
     public static CanvasDocument Load(Stream streamToMsapp, ErrorContainer errors)
     {
-        ArgumentNullException.ThrowIfNull(streamToMsapp);
+        if (streamToMsapp == null)
+        {
+            throw new ArgumentNullException(nameof(streamToMsapp));
```
Comment (joem-msft) [inline src/PAModel/Serializers/MsAppSerializer.cs:62]: nit: `_ = streamToMsapp ?? throw new ArgumentNullException(nameof(streamToMsapp));`

File: src/PAModel/Microsoft.PowerPlatform.Formulas.Tools.csproj [hunk 3]
```diff
@@ -1,6 +1,8 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
+    <TargetFramework>netstandard2.0</TargetFramework>
```
Comment (joem-msft) [inline src/PAModel/Microsoft.PowerPlatform.Formulas.Tools.csproj:4]: consider adding `net8.0` as a target (change to plural: `TargetFrameworks` even when there's only one).
this is good practice for allowing support of older frameworks, but to jump forward to latest .net when consumers support it.

Comment (joem-msft) [reply src/PAModel/Microsoft.PowerPlatform.Formulas.Tools.csproj:4]: it means that doc svr will be able to utilize the net8.0 version of the dll

**Review by joem-msft — APPROVED** [review]

File: src/PAModel/Microsoft.PowerPlatform.Formulas.Tools.csproj (hunk 3 above)
Comment (tehcrashxor) [reply src/PAModel/Microsoft.PowerPlatform.Formulas.Tools.csproj]: Good idea.  Done.

File: src/PAModel/Serializers/MsAppSerializer.cs (hunk 2 above)
Comment (tehcrashxor) [reply src/PAModel/Serializers/MsAppSerializer.cs]: done

**Review by petrochuk — APPROVED** [review]

**Pull request merged**